### PR TITLE
adapt tests to work with case insensitive comparison

### PR DESCRIPTION
**Pull request by anuragsoni**

### CODE REVIEW

File: cohttp/fuzz/fuzz_header.ml [hunk 1]
```diff
@@ -206,7 +206,7 @@ let mem_test () =
       [ headers_gen; header_name_gen ] (fun h k ->
         check_eq
           H.(mem h k)
-          List.(mem_assoc k (H.to_list h))))
+          List.(exists (fun (x, _) -> String.lowercase_ascii x = String.lowercase_ascii k) (H.to_list h))))
```
Comment (mseri) [inline cohttp/fuzz/fuzz_header.ml]: The specification and description of this test do not match the implementation. In this case, I think we should probably go for `same behaviour as List.mem_assoc after lowercasing the keys`
```
(* forall h, k. H.mem h k = List.(mem_assoc (String.lowercase_ascii k) (List.map (fun (k, v) -> (String.lowercase_ascii k, v) (H.to_list h))) *)
```

What do you think?

**Review by mseri — CHANGES_REQUESTED** [review]
Thanks a lot! I think we need to change one of those. We could just update the description, but I think it would be more useful to make a slightly different check

File: cohttp/fuzz/fuzz_header.ml (hunk 1 above)
Comment (anuragsoni) [reply cohttp/fuzz/fuzz_header.ml]: Thanks! Your suggestion is indeed a lot nicer than what I had. I've updated the PR with the new test spec + implementation.

**Comment by mseri** [comment]
Thanks a lot!

**Pull request merged**